src/Microsoft.ML.AutoML/API/ColumnInference.cs

```diff
@@ -60,6 +60,7 @@ public sealed class ColumnInformation
 
         /// <summary>
         /// The dataset column to use as a group ID for computation.
+        /// If a SamplingKeyColumnName is provided, then it should be the same as this column.
```

Comment:
nit: "... for ranking tasks". In general this option is only used for Ranking, so maybe making that explicit would be a bit more useful 😋